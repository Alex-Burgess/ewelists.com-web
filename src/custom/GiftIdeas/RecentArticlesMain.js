import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Card from "components/Card/Card.js";
import CardHeader from "components/Card/CardHeader.js";
import Info from "components/Typography/Info.js";
import Danger from "components/Typography/Danger.js";

import styles from "assets/jss/custom/components/recentArticlesMainStyle.js";
const useStyles = makeStyles(styles);

export default function RecentArticlesMain(props) {
  const { articles } = props;
  const classes = useStyles();

  const renderLeftImg = (classes, title, url, img, description_short, beginning_content) => {
    return (
      <Card plain blog className={classes.card}>
        <GridContainer>
          <GridItem xs={12} sm={5} md={5}>
            <CardHeader image plain>
              <a href={url}>
                <img src={require('assets/img/articles/' + img)} className={classes.leftImage} alt="..." />
              </a>
              <div
                className={classes.coloredShadow}
                style={{
                  backgroundImage: `url(${require('assets/img/articles/' + img)})`,
                  opacity: "1"
                }}
              />
            </CardHeader>
          </GridItem>
          <GridItem xs={12} sm={7} md={7}>
            <Info>
              <h6 className={classes.cardCategory}>
                <a href={url}>
                  {title}
                </a>
              </h6>
            </Info>
            <h3 className={classes.cardTitle}>
              <a href={url}>
                {description_short}
              </a>
            </h3>
            <p className={classes.description1}>
              {beginning_content}
              <a href={url}>
                {" "}
                Read More{" "}
              </a>
            </p>
          </GridItem>
          <GridItem xs={12} sm={5} md={5}>
            <CardHeader image plain>
              <a href={url}>
                <img src={require('assets/img/articles/' + img)} className={classes.leftImageMobile} alt="..." />
              </a>
              <div
                className={classes.coloredShadow}
                style={{
                  backgroundImage: `url(${require('assets/img/articles/' + img)})`,
                  opacity: "1"
                }}
              />
            </CardHeader>
          </GridItem>
        </GridContainer>
      </Card>
    );
  }

  const renderRightImg = (classes, title, url, img, description_short, beginning_content) => {
    return (
      <Card plain blog className={classes.card}>
          <GridContainer>
            <GridItem xs={12} sm={7} md={7}>
              <Danger>
                <h6 className={classes.cardCategory}>
                  <a href={url}>
                    {title}
                  </a>
                </h6>
              </Danger>
              <h3 className={classes.cardTitle}>
                <a href={url}>
                  {description_short}
                </a>
              </h3>
              <p className={classes.description1}>
              {beginning_content}
                <a href={url}>
                  {" "}
                  Read More{" "}
                </a>
              </p>
            </GridItem>
            <GridItem xs={12} sm={5} md={5}>
              <CardHeader image plain>
                <a href={url}>
                  <img src={require('assets/img/articles/' + img)} className={classes.rightImage} alt="..." />
                </a>
                <div
                  className={classes.coloredShadow}
                  style={{
                    backgroundImage: `url(${require('assets/img/articles/' + img)})`,
                    opacity: "1"
                  }}
                />
              </CardHeader>
            </GridItem>
          </GridContainer>
        </Card>
    );
  }

  return (
    <div className={classes.section}>
      <GridContainer>
        <GridItem xs={12} sm={10} md={10} className={classes.mlAuto + " " + classes.mrAuto}>
          {
            articles.map ((article, i) =>
              <div>
                {article.img_position_left
                  ? renderLeftImg(
                      classes,
                      article.title,
                      article.url,
                      article.img,
                      article.description_short,
                      article.beginning_content
                    )
                  : renderRightImg(
                      classes,
                      article.title,
                      article.url,
                      article.img,
                      article.description_short,
                      article.beginning_content
                  )
                }
              </div>
            )
          }
        </GridItem>
      </GridContainer>
    </div>
  );
}

RecentArticlesMain.propTypes = {
  articles: PropTypes.array,
};
