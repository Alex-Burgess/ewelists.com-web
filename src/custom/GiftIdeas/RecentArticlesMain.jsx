import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import withStyles from "@material-ui/core/styles/withStyles";
import GridContainer from "components/Grid/GridContainer.jsx";
import GridItem from "components/Grid/GridItem.jsx";
import Card from "components/Card/Card.jsx";
import CardHeader from "components/Card/CardHeader.jsx";
import Info from "components/Typography/Info.jsx";
import Danger from "components/Typography/Danger.jsx";

import articlesStyle from "assets/jss/material-kit-pro-react/custom/recentArticlesMainStyle.jsx";

class RecentArticlesMain extends React.Component {
  renderLeftImg(classes, title, url, img, description_short, beginning_content) {
    return (
      <Card plain blog className={classes.card}>
        <GridContainer>
          <GridItem xs={12} sm={5} md={5}>
            <CardHeader image plain>
              <a href={url}>
                <img src={require('assets/img/articles/' + img)} className={classes.leftImage} alt="..." />
              </a>
              <div
                className={classes.coloredShadow}
                style={{
                  backgroundImage: `url(${require('assets/img/articles/' + img)})`,
                  opacity: "1"
                }}
              />
            </CardHeader>
          </GridItem>
          <GridItem xs={12} sm={7} md={7}>
            <Info>
              <h6 className={classes.cardCategory}>
                <a href={url}>
                  {title}
                </a>
              </h6>
            </Info>
            <h3 className={classes.cardTitle}>
              <a href={url}>
                {description_short}
              </a>
            </h3>
            <p className={classes.description1}>
              {beginning_content}
              <a href={url}>
                {" "}
                Read More{" "}
              </a>
            </p>
          </GridItem>
          <GridItem xs={12} sm={5} md={5}>
            <CardHeader image plain>
              <a href={url}>
                <img src={require('assets/img/articles/' + img)} className={classes.leftImageMobile} alt="..." />
              </a>
              <div
                className={classes.coloredShadow}
                style={{
                  backgroundImage: `url(${require('assets/img/articles/' + img)})`,
                  opacity: "1"
                }}
              />
            </CardHeader>
          </GridItem>
        </GridContainer>
      </Card>
    );
  }

  renderRightImg(classes, title, url, img, description_short, beginning_content) {
    return (
      <Card plain blog className={classes.card}>
          <GridContainer>
            <GridItem xs={12} sm={7} md={7}>
              <Danger>
                <h6 className={classes.cardCategory}>
                  <a href={url}>
                    {title}
                  </a>
                </h6>
              </Danger>
              <h3 className={classes.cardTitle}>
                <a href={url}>
                  {description_short}
                </a>
              </h3>
              <p className={classes.description1}>
              {beginning_content}
                <a href={url}>
                  {" "}
                  Read More{" "}
                </a>
              </p>
            </GridItem>
            <GridItem xs={12} sm={5} md={5}>
              <CardHeader image plain>
                <a href={url}>
                  <img src={require('assets/img/articles/' + img)} className={classes.rightImage} alt="..." />
                </a>
                <div
                  className={classes.coloredShadow}
                  style={{
                    backgroundImage: `url(${require('assets/img/articles/' + img)})`,
                    opacity: "1"
                  }}
                />
              </CardHeader>
            </GridItem>
          </GridContainer>
        </Card>
    );
  }

  render() {
    const { classes, articles } = this.props;

    return (
      <div className={classes.section}>
        <GridContainer>
          <GridItem xs={12} sm={10} md={10} className={classes.mlAuto + " " + classes.mrAuto}>
            {
              articles.map ((article, i) =>
                <div>
                  {article.img_position_left
                    ? this.renderLeftImg(
                        classes,
                        article.title,
                        article.url,
                        article.img,
                        article.description_short,
                        article.beginning_content
                      )
                    : this.renderRightImg(
                        classes,
                        article.title,
                        article.url,
                        article.img,
                        article.description_short,
                        article.beginning_content
                    )
                  }
                </div>
              )
            }
          </GridItem>
        </GridContainer>
      </div>
    );
  }
}

RecentArticlesMain.propTypes = {
  classes: PropTypes.object.isRequired,
  articles: PropTypes.array,
};

export default withStyles(articlesStyle)(RecentArticlesMain);
