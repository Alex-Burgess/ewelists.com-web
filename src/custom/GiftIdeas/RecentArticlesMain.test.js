import * as React from 'react';
import renderer from 'react-test-renderer';
import RecentArticlesMain from './RecentArticlesMain';

test('Renders RecentArticles component.', () => {
  const recentArticles = [
    {title: "Christmas Ideas", url: "/list-ideas/christmas-ideas-for-toddlers", img: 'christmastoddlers.jpg', img_position_left: false,
    description_short: "Great Christmas ideas for toddlers and young children.",
    beginning_content: "If you are stuck wondering what to get a toddler for Christmas, here are some of our favourite ideas! Toddlers love to mimic adults and join in ‘helping’ with every day tasks. Here are some great ideas to encourage role play..."},
    {title: "Travel Gear", url: "/list-ideas/baby-travel-gear", img: 'baby-travel-gear.jpg', img_position_left: true,
    description_short: "Great items to make travelling with your little ones no fuss!",
    beginning_content: "The compact stroller/buggy is one of the most useful items for travelling. They are handy for all sorts, such as effortlessly moving around the city hopping on and off public transport, taking with you through the airport, or just keeping in the car..."}
  ];

  const tree = renderer
    .create(
      <RecentArticlesMain
        articles={
          recentArticles
        }
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
