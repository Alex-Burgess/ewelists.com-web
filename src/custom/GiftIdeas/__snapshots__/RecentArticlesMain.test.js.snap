// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders RecentArticles component. 1`] = `
<div
  className="makeStyles-section-3"
>
  <div
    className="MuiGrid-root makeStyles-grid-10  MuiGrid-container"
  >
    <div
      className="MuiGrid-root makeStyles-grid-114 makeStyles-mlAuto-1 makeStyles-mrAuto-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10 MuiGrid-grid-md-10"
    >
      <div>
        <div
          className="makeStyles-card-115 makeStyles-cardPlain-116 makeStyles-cardBlog-118"
        >
          <div
            className="MuiGrid-root makeStyles-grid-10  MuiGrid-container"
          >
            <div
              className="MuiGrid-root makeStyles-grid-114  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-7 MuiGrid-grid-md-7"
            >
              <div
                className="makeStyles-defaultFontStyle-130 makeStyles-dangerText-140"
              >
                <h6
                  className="makeStyles-cardCategory-5"
                >
                  <a
                    href="/listideas/christmasfortoddlers"
                  >
                    Christmas Ideas
                  </a>
                </h6>
              </div>
              <h3
                className="makeStyles-cardTitle-4"
              >
                <a
                  href="/listideas/christmasfortoddlers"
                >
                  Great Christmas ideas for toddlers and young children.
                </a>
              </h3>
              <p
                className="makeStyles-description1-6"
              >
                If you are stuck wondering what to get a toddler for Christmas, here are some of our favourite ideas! Toddlers love to mimic adults and join in ‘helping’ with every day tasks. Here are some great ideas to encourage role play...
                <a
                  href="/listideas/christmasfortoddlers"
                >
                   
                  Read More
                   
                </a>
              </p>
            </div>
            <div
              className="MuiGrid-root makeStyles-grid-114  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-5"
            >
              <div
                className="makeStyles-cardHeader-143 makeStyles-cardHeaderPlain-144 makeStyles-cardHeaderImage-145"
              >
                <a
                  href="/listideas/christmasfortoddlers"
                >
                  <img
                    alt="..."
                    className="makeStyles-rightImage-9"
                    src="christmastoddlers.jpg"
                  />
                </a>
                <div
                  style={
                    Object {
                      "backgroundImage": "url(christmastoddlers.jpg)",
                      "opacity": "1",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="makeStyles-card-115 makeStyles-cardPlain-116 makeStyles-cardBlog-118"
        >
          <div
            className="MuiGrid-root makeStyles-grid-10  MuiGrid-container"
          >
            <div
              className="MuiGrid-root makeStyles-grid-114  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-5"
            >
              <div
                className="makeStyles-cardHeader-143 makeStyles-cardHeaderPlain-144 makeStyles-cardHeaderImage-145"
              >
                <a
                  href="/listideas/travelgear"
                >
                  <img
                    alt="..."
                    className="makeStyles-leftImage-7"
                    src="travelgear.jpg"
                  />
                </a>
                <div
                  style={
                    Object {
                      "backgroundImage": "url(travelgear.jpg)",
                      "opacity": "1",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="MuiGrid-root makeStyles-grid-114  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-7 MuiGrid-grid-md-7"
            >
              <div
                className="makeStyles-defaultFontStyle-155 makeStyles-infoText-162"
              >
                <h6
                  className="makeStyles-cardCategory-5"
                >
                  <a
                    href="/listideas/travelgear"
                  >
                    Travel Gear
                  </a>
                </h6>
              </div>
              <h3
                className="makeStyles-cardTitle-4"
              >
                <a
                  href="/listideas/travelgear"
                >
                  Great items to make travelling with your little ones no fuss!
                </a>
              </h3>
              <p
                className="makeStyles-description1-6"
              >
                The compact stroller/buggy is one of the most useful items for travelling. They are handy for all sorts, such as effortlessly moving around the city hopping on and off public transport, taking with you through the airport, or just keeping in the car...
                <a
                  href="/listideas/travelgear"
                >
                   
                  Read More
                   
                </a>
              </p>
            </div>
            <div
              className="MuiGrid-root makeStyles-grid-114  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-5"
            >
              <div
                className="makeStyles-cardHeader-143 makeStyles-cardHeaderPlain-144 makeStyles-cardHeaderImage-145"
              >
                <a
                  href="/listideas/travelgear"
                >
                  <img
                    alt="..."
                    className="makeStyles-leftImageMobile-8"
                    src="travelgear.jpg"
                  />
                </a>
                <div
                  style={
                    Object {
                      "backgroundImage": "url(travelgear.jpg)",
                      "opacity": "1",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
